### PR TITLE
BUG: Remove obsolete VRG Generation functionality

### PR DESCRIPTION
**Pull request by sbelsk**
Closes https://github.com/BrownBiomechanics/SlicerAutoscoperM/issues/132.

This PR removes the 'Manual Camera Placement' and 'Automatic Camera Placement' VRG Generation UI groups from the Pre-Processing modules, along with all of their associated backend code. This includes the removal of the `CalculateDataIntensityDensity` and `VirtualRadiographGeneration` modules.
The support for camera placement is superseded by the hierarchical registration capability.

### CODE REVIEW

File: AutoscoperM/AutoscoperMLib/IO.py [hunk 1]
```diff
@@ -265,7 +235,7 @@ def writeTRA(fileName: str, transforms: list[vtk.vtkMatrix4x4]) -> None:
             traFile.write(",".join(row) + "\n")
 
 
-def writeTemporyFile(filename: str, data: vtk.vtkImageData) -> str:
+def writeTemporaryFile(filename: str, data: vtk.vtkImageData) -> str:
```
Comment (jcfr) [inline AutoscoperM/AutoscoperMLib/IO.py:238]: I suggest to fix the typos in `writeTemporyFile` and `removeTemporyFile` in its own pull request (either before or after this one)

File: AutoscoperM/AutoscoperMLib/IO.py [hunk 2]
```diff
@@ -35,34 +33,6 @@ def loadSegmentation(segmentationNode: slicer.vtkMRMLSegmentationNode, filename:
         return None
 
 
-def generateCameraCalibrationFile(camera: Camera, filename: str):
-    """
```
Comment (jcfr) [inline AutoscoperM/AutoscoperMLib/IO.py:39]: Once this is integrated, let's also make sure to submit a pull request against `BrownBiomechanics/Autoscoper` to add a `deprecated` admonition to https://github.com/BrownBiomechanics/Autoscoper/blob/main/Documentation/file-specifications/camera-calibration.md#overview

**Review by jcfr — COMMENTED** [review]
Thanks for working on those changes :pray: 

Beside of few minor nitpicks, it will be good for integration

In the commit message, could you also provide the rational for removing the removing the features ?  (e.g superseded by the integration of hierarchical registration allowing to ...)

**Review by amymmorton — APPROVED** [review]
LGTM

image

File: AutoscoperM/AutoscoperMLib/IO.py (hunk 1 above)
Comment (sbelsk) [reply AutoscoperM/AutoscoperMLib/IO.py]: > these functions are actually no longer being used.

To reduce technical debt, I removed these two functions.

> hmmm.. what is used in the #99 ? When registration are output?

It seems that when the "Export Transforms" button is clicked in the 3DHierarchicalRegistration module, the function [`AutoscoperM/AutoscoperMLib/IO.py:writeTRA(fileName, transforms)`](https://github.com/sbelsk/SlicerAutoscoperM/blob/remove-VRG-generation/AutoscoperM/AutoscoperMLib/IO.py#L229) is used (not affected by this PR).

File: AutoscoperM/AutoscoperMLib/IO.py (hunk 2 above)
Comment (sbelsk) [reply AutoscoperM/AutoscoperMLib/IO.py:39]: Opened PR https://github.com/BrownBiomechanics/Autoscoper/pull/307 to update the documentation.

**Review by jcfr — APPROVED** [review]
Thanks for addressing the comments :pray: 

I suggest to further tweak the commit message and pull request description to include the rational. Feel free to tweak/adapt as you see fit.

```diff
Removes the 'Manual Camera Placement' and 'Automatic Camera Placement'
VRG Generation UI groups from the Pre-Processing modules, along with all
of their associated backend code. This includes the removal of the
CalculateDataIntensityDensity and VirtualRadiographGeneration modules.
+ The camera placement capabilities is superseded by the hierarchical registration capabilities.
```

**Pull request merged**